cache/key_test.go

```diff
@@ -28,10 +32,107 @@ func TestWithVersionSuffix(t *testing.T) {
 			},
 			want: "sha256:1acfb4a0eb0acb0b1e8d462df18841581b960d5df190b32f3327df4820e5bc7b",
 		},
+		{
+			name: "empty slice file patterns",
+			args: args{
+				key:     "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e",
+				version: "111101112110013",
+			},
+			patterns: []string{},
+			want:     "sha256:1acfb4a0eb0acb0b1e8d462df18841581b960d5df190b32f3327df4820e5bc7b",
+		},
+		{
+			name: "empty string in file patterns",
+			args: args{
+				key:     "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e",
+				version: "111101112110013",
+			},
+			patterns: []string{""},
+			want:     "sha256:1acfb4a0eb0acb0b1e8d462df18841581b960d5df190b32f3327df4820e5bc7b",
+		},
+		{
+			name: "multiple empty string in file patterns",
+			args: args{
+				key:     "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e",
+				version: "111101112110013",
+			},
+			patterns: []string{"", ""},
+			want:     "sha256:1acfb4a0eb0acb0b1e8d462df18841581b960d5df190b32f3327df4820e5bc7b",
+		},
+		{
+			name: "non empty string in file patterns",
+			args: args{
+				key:     "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e",
+				version: "111101112110013",
+			},
+			patterns: []string{"test"},
+			want:     "sha256:7bbca3f56a19cc7d892879d17ba2204f62d37fd9122ca74271e5bb4ab0817bc1",
+		},
+		{
+			name: "empty and non empty string in file patterns",
+			args: args{
+				key:     "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e",
+				version: "111101112110013",
+			},
+			patterns: []string{"test", ""},
+			want:     "sha256:7bbca3f56a19cc7d892879d17ba2204f62d37fd9122ca74271e5bb4ab0817bc1",
+		},
+		{
+			name: "empty slice policy paths",
+			args: args{
+				key:     "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e",
+				version: "111101112110013",
+			},
+			policy: []string{},
+			want:   "sha256:1acfb4a0eb0acb0b1e8d462df18841581b960d5df190b32f3327df4820e5bc7b",
+		},
+		{
```

Comment:
I have yet to find what []string{""} is evaluated too, but it probably depends on dir content too.

Reply:
I'm guessing `[]string{""}` is the same as `[]string{"."}`.

Reply:
oh I misread and thought they produced different hashes, that's why I took note :facepalm: removing as well